src/main/java/com/scaleunlimited/flinkcrawler/warc/WARCFileWriter.java

```diff
@@ -0,0 +1,219 @@
+package com.scaleunlimited.flinkcrawler.warc;
+
+import java.io.BufferedOutputStream;
+import java.io.DataOutputStream;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.util.Progressable;
+import org.apache.hadoop.util.ReflectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Writes {@link WARCRecord}s to a WARC file, using Hadoop's filesystem APIs. (This means you can
+ * write to HDFS, S3 or any other filesystem supported by Hadoop). This implementation is not tied
+ * to the MapReduce APIs -- that link is provided by the mapred
+ * {@link com.martinkl.warc.mapred.WARCOutputFormat} and the mapreduce
+ * {@link com.martinkl.warc.mapreduce.WARCOutputFormat}.
+ *
+ * WARCFileWriter keeps track of how much data it has written (optionally gzip-compressed); when the
+ * file becomes larger than some threshold, it is automatically closed and a new segment is started.
+ * A segment number is appended to the filename for that purpose. The segment number always starts
+ * at 00000, and by default a new segment is started when the file size exceeds 1GB. To change the
+ * target size for a segment, you can set the `warc.output.segment.size` key in the Hadoop
+ * configuration to the number of bytes. (Files may actually be a bit larger than this threshold,
+ * since we finish writing the current record before opening a new file.)
+ */
+public class WARCFileWriter {
+    private static final Logger logger = LoggerFactory.getLogger(WARCFileWriter.class);
```

Comment:
I'm wondering why here you're not using our naming conventions for constants (i.e., LOGGER). Did this code come from elsewhere, but you didn't include any attribution?

Reply:
I didn't add any additional attribution - just included the file as-is... Should I add attribution and then go ahead and make changes to conform to our usual guidelines ?

Reply:
I recommend adding attribution to the file and otherwise leaving the code as is.